libplatsupport/src/mach/eswin/timer.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2025, UNSW
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <platsupport/mach/timer.h>
+#include <stdint.h>
+#include <utils/util.h>
+
+#define MAX_TIMEOUT_NS (ESWIN_TIMER_MAX_TICKS * (NS_IN_S / ESWIN_TIMER_TICKS_PER_SECOND))
+
+void eswin_timer_enable(eswin_timer_t *timer)
+{
+    assert(timer);
+    assert(timer->regs);
+
+    timer->regs->ctrl |= 0x1;
+}
+
+void eswin_timer_disable(eswin_timer_t *timer)
+{
+    assert(timer);
+    assert(timer->regs);
+
+    timer->regs->ctrl &= 0xfffffffe;
+}
+
+void eswin_timer_handle_irq(eswin_timer_t *timer)
+{
+    assert(timer);
+    assert(timer->regs);
+
+    timer->value_h += 1;
+
+    /* Reading the EIO register acks the IRQ */
+    timer->regs->eoi;
+}
+
+uint64_t eswin_timer_get_time(eswin_timer_t *timer)
+{
+    assert(timer);
+    assert(timer->regs);
+
+    /* the timer value counts down from the load value */
+    uint64_t value_l = (uint64_t)(ESWIN_TIMER_MAX_TICKS - timer->regs->value);
```

Comment:
Is this right? Does the value in `load_count` count affect `value`? That would make the timer unsuitable for a combined timer and clock.

Usually `value` would increment each tick and on overflow you get the overflow interrupt, but `value` would be free running compared to `load_count`.

Edit: never mind, I see you're using two timers because of this. Worth a comment somewhere.